server/server.go

```diff
@@ -362,3 +364,51 @@ func (s *Server) startGarbageCollection(ctx context.Context, frequency time.Dura
 	}()
 	return
 }
+
+// getUpdatedConnectorList retrieves a current list of connector objects available in
+// the storage. It gets the latest configuration of connectors that are dynamically
+// updated via the API.
+func (s *Server) getUpdatedConnectorList() error {
```

Comment:
This should probably just return the list of connectors. It feels weird to have to call this before accessing the map.